### PR TITLE
[MISC] Stop autobrake max disengaging on the ground

### LINKED ISSUE #2586: AUTOBREAK MAX turn off while taxing/breaking before T/O
**Mod Version**

MASTER (FBW)

**Describe the bug**

when i prepare the A/C for Takeoff after engine start and press "autobreak MAX", the preselected autobreak turnes off while taxing to the runway. Eighter when breaking or only for no reasons. it only be active for few secounds.

**To Reproduce**

1. prepare the Aircraft from Cold and dark for takeoff
2. set autobreak MAX for takeoff
3.

**Expected behavior**

autobreak MAX shoud stay on/active until Takeoff 

**Actual behavior**

autobreak MAX turn off, eighter when breaking on the way to the runway, or by itself after few secounds

**References**

-

**Additional context**

-
Was this working before/when did the issue start occurring?
worked befor 0.5.0
Is this a problem in the vanilla unmodded game?
no

Discord username (if different from GitHub):

### PR DESCRIPTION
**Pull request by pareil6**
Fixes #2586.

## Summary of Changes
Fixes autobrake max disengaging after a short period of time while on the ground, caused by a subtle logic error introduced in the RNP conversion.

## Screenshots (if necessary)
N/A

## References
N/A

## Additional context
N/A

Discord username (if different from GitHub): pareil6#0990

## Testing instructions
Try setting autobrake max, try taxiing, using the foot brakes and checking that doesn't reset it, take off and check it clears after the landing gear comes up, check it can't be selected once in the air.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Review by Benjozork — APPROVED** [review]
LGTM

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 1. [CDU] Improved ascent constraint algorithm - @MisterChocker (Leon)
 1. [CDU] Fixed FPLN airways showing as undefined - @pareil6 - (pareil6)
 1. [CDU] Allow SimBrief OFP to override previous flight plan - @pareil6 (pareil6)
+1. [MISC] Stop autobrake max disengaging on the ground - @pareil6 (pareil6)
```
Comment (pepperoni505) [inline .github/CHANGELOG.md:15]: Changelog isn't needed

**Comment by valenzz2004** [comment]
Category: QA Tester
Name: Valenzz2004#7802
Date of testing: xx/12/2020
Version of the sim: 1.12.13.0
PR Tested: #2621 
Object of test: Stop autobrake max disengaging on the ground

Testing technique: Laps taxiing at egll
Steps you've made to test the Pull Request:

1. Loaded c/d 
2. set up aircraft for pattern work 
3. pushed back and set all TO config settings as in an ifr flight
4. taxied TWY A for 2 laps with 6 brake tests/checks
5.  two RTO's rwy 27L 

Flight notes (If required): Pattern Work setup

Issues (If Any): during RTO autobrake max did not engage while spoilers deployed as expected so small issue
Overall Rating: Small improvement needed
Conclusions of the testing: ready to go and found no issues while doing 2 laps of EGLL outer taxiway and neither parking brake nor toe brake disengage Autobrake max. 
Screenshots( if any):

Before Take-off lined up
b4 RTO

During RTO max reverse and spoilers deployed
during rto

After RTO (autobrake max is still armed)
after RTO

**Comment by NickFarin** [comment]
Category: QA Tester
Name: Discord User: Nick F[Z+8] #5327
Date of testing: Current date 23/12/2020
Version of the sim: 1.12.13.0
PR Tested: #2621 
Object of test: (ECAM Page or Flight model or views or anything, what the PR modifies)

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1.  Spawned at KLAS in cold and dark
2.  Started the engines and pushed back
3.  Started taxi to runway and apply brakes many times
4.  Performed RTO to see if the autobrake is still on after RTO
5. Performed takeoff to see if autobrakes disengages after takeoff
5. Performed landing to see if autobrakes will show decel and if it will disengage after brake pressure is applied

Flight notes (If required):

Issues (If Any): 
Overall Rating: Good
Conclusions of the testing: Test went normal. Autobrakes do not disengage when on ground.

**Comment by pareil6** [comment]
Checking failed RTO cases now, either another logic bug snuck in during the RNP update or this was broken in v0.5.0 and we hadn't realised...

**Comment by NickFarin** [comment]
Category: QA Trial
Name: Discord User: Nick F[Z+8] #5327
Date of testing: Current date 23/12/2020
Version of the sim: 1.12.13.0
PR Tested: #2621
Object of test: autobrake

Testing technique: apply brake pressure during taxi 
Steps you've made to test the Pull Request:

1.Spawned at KLAS in cold and dark
2.Started the engines and pushed back
3.Started taxi to runway and apply brakes many times
4.Performed RTO to see if the autobrake is still on after RTO
5.Performed takeoff to see if autobrakes disengages after takeoff
6.Performed landing to see if autobrakes will show decel and if it will disengage after brake pressure is applied
7.Flight notes (If required):

Issues (If Any):
Overall Rating: Good
Conclusions of the testing: Test went normal. Autobrakes do not disengage when on ground.

**Comment by ImenesFBW** [comment]
Should be good to merge: @Benjozork @pepperoni505 

RTO behavior is same on stable and dev. Possible update issue?

**Comment by pareil6** [comment]
Just for completeness - tested the stable version by backporting the old RPN ModelBehaviourDefs into current dev version, and the RTO behaviour is still broken. Can't see any other changes that would affect autobrake behaviour, so assuming this must have been there before (or was broken by Asobo rather than us in the sim update).

**Comment by marinerhally** [comment]
Date of testing: Current date 22/12/2020
Version of the sim: 1.12.13.0
PR Tested: #2621
Object of test: Stop Autobrake disengaging.

Testing technique: Spawned C/D taxied to runway then aborted t/o.
Steps you've made to test the Pull Request:
1. Spawned C/D
2. While setting up plane set max auto brake
3. Checked max auto brake was still engaged
4. While taxing kept hitting the brakes (max brake still engaged).
5. On take off at 110kts aborted t/o (max brake still engaged)
6. Repeated steps 1-5
7. Then did 2 x take offs spawning on runway.

Flight notes (If required):

Issues (If Any): None
Overall Rating: Good
Conclusions of the testing: Max brake stayed on at all times. Well done to the Devs great work.

**Comment by ErickSharp** [comment]
Category: QA Trial
Name: 🎄🎅Erick🎄🎅 [Z-5]#6484
Date of testing: Current date 22/12/2020
Version of the sim: 1.12.13.0
PR Tested: #2621
Object of test: Autobrake max disengagement issues

Testing technique: Tested the autobrake issues on the development version, and the PR meant to address the issue.
Steps you've made to test the Pull Request:

Development version:
https://youtu.be/S4nVnf4okUo

PR:
https://youtu.be/GXRdTTaSX_4

Flight notes (If required): Madeira airport for all testing.
Issues (If Any): Autobrake does not activate on a rejected takeoff. 
Overall Rating: Good/Might need some improvement
Conclusions of the testing: Although during RTO, the autobrake does not activate, the PR does fix the issue of the autobrake deactivating on the ground. May need some more improvement or could be merged and the issue of autobrake not activating on a rejected takeoff could be handled in another PR.

**Comment by mikecbr** [comment]
guys , the issue still there , i just checke it.....

**Comment by pepperoni505** [comment]
@mikecbr The FBW Branch isn't always current, but the fix should be merged later today or tomorrow.

**Comment by mikecbr** [comment]
> @mikecbr The FBW Branch isn't always current, but the fix should be merged later today or tomorrow.

Okey thanks for the reply , 💯

**Pull request merged**